packages/jest-resolve-dependencies/src/index.ts

```diff
@@ -64,8 +67,31 @@ class DependencyResolver {
         }
       }
 
-      if (resolvedDependency) {
-        acc.push(resolvedDependency);
+      if (!resolvedDependency) {
+        return acc;
+      }
+
+      acc.push(resolvedDependency);
+
+      // If we resolve a dependency, then look for a mock dependency
+      // of the same name in that dependency's directory.
+      resolvedMockDependency = this._resolver.getMockModule(
+        resolvedDependency,
+        path.basename(dependency),
+      );
+
+      if (resolvedMockDependency) {
+        const dependencyMockDir = path.resolve(
+          path.dirname(resolvedDependency),
+          '__mocks__',
+        );
+
+        resolvedMockDependency = path.resolve(resolvedMockDependency);
+
+        // make sure mock is in the correct directory
+        if (dependencyMockDir === path.dirname(resolvedMockDependency)) {
```

Comment:
Is there a test that fails if this `acc.push` is unconditional?

Reply:
Yes! Both tests fail if this gets removed. It would basically add an extra non-valid dependency to the list